pkg/fanal/utils/utils.go

```diff
@@ -93,3 +94,31 @@ func IsBinary(content xio.ReadSeekerAt, fileSize int64) (bool, error) {
 
 	return false, nil
 }
+
+func ExtractPrintableBytes(content xio.ReadSeekerAt) ([]byte, error) {
+	var printalbe []byte
+	current := make([]byte, 1)
+
+	wasReadable := false
+
+	for {
+		_, err := content.Read(current)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, err
+		}
+		if unicode.IsPrint(rune(current[0])) {
+			if !wasReadable {
+				printalbe = append(printalbe, byte(' '))
+				wasReadable = true
+			}
+			printalbe = append(printalbe, current[0])
```

Comment:
This implementation concats all single printable characters, leading to the long string. Even in a binary file, there are many printable character strings when viewed on a per-byte basis. I think the `strings` way, setting the minimum length, is better like [I shared](https://github.com/aquasecurity/trivy/issues/7204).